nilearn/_utils/data_gen.py

```diff
@@ -866,91 +878,427 @@ def create_fake_bids_dataset(base_dir='',
         Creates a directory with dummy files.
 
     """
-    bids_path = os.path.join(base_dir, 'bids_dataset')
-    os.makedirs(bids_path)
+    n_voxels = 4
+
     rand_gen = check_random_state(random_state)
-    # Create surface bids dataset
-    open(os.path.join(bids_path, 'README.txt'), 'w')
-    vox = 4
-    created_sessions = ['ses-%02d' % label for label in range(1, n_ses + 1)]
-    if no_session:
-        created_sessions = ['']
-    for subject in ['sub-%02d' % label for label in range(1, n_sub + 1)]:
-        for session in created_sessions:
-            subses_dir = os.path.join(bids_path, subject, session)
-            if session in ('ses-01', ''):
-                anat_path = os.path.join(subses_dir, 'anat')
-                os.makedirs(anat_path)
-                anat_file = os.path.join(anat_path, subject + '_T1w.nii.gz')
-                open(anat_file, 'w')
-            func_path = os.path.join(subses_dir, 'func')
-            os.makedirs(func_path)
-            for task, n_run in zip(tasks, n_runs):
-                run_labels = [
-                    'run-%02d' % label for label in range(1, n_run + 1)
-                ]
-                for run in run_labels:
-                    fields = [subject, session, 'task-' + task]
-                    if '' in fields:
-                        fields.remove('')
-                    file_id = '_'.join(fields)
-                    if n_run > 1:
-                        file_id += '_' + run
-                    bold_path = os.path.join(func_path,
-                                             file_id + '_bold.nii.gz')
-                    write_fake_bold_img(bold_path, [vox, vox, vox, 100],
-                                        random_state=rand_gen)
-                    events_path = os.path.join(func_path,
-                                               file_id + '_events.tsv')
-                    basic_paradigm().to_csv(events_path, sep='\t', index=None)
-                    param_path = os.path.join(func_path,
-                                              file_id + '_bold.json')
-                    with open(param_path, 'w') as param_file:
-                        json.dump({'RepetitionTime': 1.5}, param_file)
-
-    # Create derivatives files
+
+    bids_dataset_dir = 'bids_dataset'
+
```

Comment:
TODO: add some input validation of entities to make sure BIDS labels are not only strings but also alphanumeric.